src/drivers/serial/raspi_uart.c

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file
+ * @brief Implements serial driver for Raspberry PI
+ *
+ * See BCM2835-ARM-Peripherals.pdf, 13 chapter for details.
+ *
+ * @date 02.07.15
+ * @author Vita Loginova
+ */
+
+#include <hal/reg.h>
+#include <stdint.h>
+#include <drivers/diag.h>
+
+#define BAUD_RATE OPTION_GET(NUMBER, baud_rate) /* default 115200 */
+#define UART_CLOCK 3000000
+#define UART_DIVIDER UART_CLOCK/(16 * BAUD_RATE)
+#define UART_FRACTIONAL 40 /* (.627 * 64) + 0.5 */
+
+#define GPIO_OFFSET  0x20200000
+
+/* Controls actuation of pull up/down to ALL GPIO pins. */
+#define GPPUD        GPIO_OFFSET + 0x94
+
+/* Controls actuation of pull up/down for specific GPIO pin. */
+#define GPPUDCLK0    GPIO_OFFSET + 0x98
+
+#define UART0_OFFSET GPIO_OFFSET + 0x00001000
+
+#define UART0_DR     UART0_OFFSET + 0x00 /* Data Register  */
+#define UART0_FR     UART0_OFFSET + 0x18 /* Flag register */
+#define UART0_IBRD   UART0_OFFSET + 0x24 /* Integer Baud rate divisor */
+#define UART0_FBRD   UART0_OFFSET + 0x28 /* Fractional Baud rate divisor */
+#define UART0_LCRH   UART0_OFFSET + 0x2C /* Line Control register */
+#define UART0_CR     UART0_OFFSET + 0x30 /* Control register */
+#define UART0_IMSC   UART0_OFFSET + 0x38 /* Interrupt Mask Set Clear Register */
+#define UART0_ICR    UART0_OFFSET + 0x44 /* Interrupt Clear Register */
+
+
+static void delay(int32_t count) {
```

Comment:
Assembler for delay routine in not very good idea. Can it be changed to more suit version?